argilla/docs/how_to_guides/custom_fields.md

```diff
@@ -0,0 +1,225 @@
+    display: flex;
+    gap: 10px;
+}
+.column {
+    flex: 1;
+}
+</style>
+""" # (1)
+
+html_template = """
+<div id="container">
+    <div class="column">
+        <h3>Original</h3>
+        <img src="{{record.fields.image.original}}" />
+    </div>
+    <div class="column">
+        <h3>Revision</h3>
+        <img src="{{record.fields.image.revision}}" />
+    </div>
+</div>
+""" # (2)
+```
+
+1. This is a CSS template, which ensures that the container and columns are styled.
+2. This is an HTML template, which creates a `container` with two columns and injects the value corresponding to the `key` of the `image` field into it.
+
+We can now pass these templates to the `CustomField` class.
+
+```python
+import argilla as rg
+
+custom_field = rg.CustomField(
+    name="image",
+    template=css_template + html_template,
+)
+
+settings = rg.Settings(
+    fields=[custom_field],
+    questions=[rg.TextQuestion(name="response")],
+)
+
+dataset = rg.Dataset(
+    name="custom_field_dataset",
+    settings=settings,
+).create()
+
+dataset.records.log([
+    rg.Record(
+        fields={
+            "image": {
+                "original": "https://argilla.io/brand-assets/argilla/argilla-logo-color-black.png",
+                "revision": "https://argilla.io/brand-assets/argilla/argilla-logo-black.png",
+            }
+        }
+    )]
+)
+```
+
+The result will be the following:
+
+![example-gallery-end](../assets/images/how_to_guides/custom_field/images_in_two_columns.png)
+
+### Example Gallery
+
+??? "Metadata in a table"
+
+    You can make it easier to read metadata by displaying it in a table. This uses handlebars to iterate over the metadata object and display each key-value pair in a row.
+
+    ```python
+    template = """
+    <style>
+        .container {
+            border: 1px solid #ddd;
+            font-family: sans-serif;
+        }
+        .row {
+            display: flex;
+            border-bottom: 1px solid #ddd;
+        }
+        .row:last-child {
+            border-bottom: none;
+        }
+        .column {
+            flex: 1;
+            padding: 8px;
+        }
+        .column:first-child {
+            border-right: 1px solid #ddd;
+        }
+    </style>
+    <div class="container">
+        <div class="header">
+            <div class="column">Metadata</div>
+            <div class="column">Value</div>
+        </div>
+        {{#each record.metadata}}
+        <div class="row">
+            <div class="column">{{@key}}</div>
+            <div class="column">{{this}}</div>
+        </div>
+        {{/each}}
+    </div>
+    """
+    record = rg.Record(
+        fields={"text": "hello"},
+        metadata={
+            "name": "John Doe",
+            "age": 25,
+        }
+    )
+    ```
+    ![example-gallery-end](../assets/images/how_to_guides/custom_field/metadata_table.png)
+
+??? "JSON viewer"
+
+    If you want to display a JSON object, you can use the `json` helper to convert the object to a string.
```

Comment:
I think we might add a bit of context here. Maybe we we can also guide people to using their own helpers or how this relies on handlebars under the hood?

Reply:
I'll add some links and extra context.